src/components/EmojiPicker/EmojiPickerMenu/index.native.js

```diff
@@ -129,18 +206,55 @@ class EmojiPickerMenu extends Component {
 
     render() {
         return (
-            <View style={styles.emojiPickerContainer}>
+            <View style={[styles.emojiPickerContainer]}>
+                <View style={[styles.pt4, styles.ph4, styles.pb1]}>
+                    <RNTextInput
+                        placeholder={this.props.translate('common.search')}
+                        placeholderTextColor={themeColors.textSupporting}
+                        onChangeText={this.filterEmojis}
+                        style={[
+                            styles.textInput,
+                            this.state.isFocused && styles.borderColorFocus,
+                        ]}
+                        onFocus={() => this.setState({isFocused: true})}
+                        onBlur={() => this.setState({isFocused: false})}
+                    />
+                </View>
+
                 <FlatList
+                    keyboardShouldPersistTaps="handled"
                     data={this.emojis}
                     renderItem={this.renderItem}
-                    keyExtractor={item => (`emoji_picker_${item.code}`)}
+                    keyExtractor={item => `emoji_picker_${item.code}`}
                     numColumns={this.numColumns}
                     style={[
                         styles.emojiPickerList,
                         this.isMobileLandscape() && styles.emojiPickerListLandscape,
                     ]}
                     stickyHeaderIndices={this.unfilteredHeaderIndices}
                 />
+                {this.state.value && (
```

Comment:
I'm a bit confused here, why are we adding a second flatlist?

Reply:
The process of rendering the full emoji sheet after clearing the search string can have performance implications, especially if many emojis are involved. So I maintain a pre-rendered list of all emojis and only render a smaller search list on top. This approach could be unnecessary overhead, I can remove it if you want

Reply:
Yeah, I don't think that's particularly necessary. Let's just do everything in one flatlist

Reply:
👍🏻